integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateExternalTable.scala

```diff
@@ -62,6 +63,31 @@ class TestCreateExternalTable extends QueryTest with BeforeAndAfterAll {
     assert(new File(originDataPath).exists())
   }
 
+
+  test("create external table with existing files2") {
```

Comment:
Suggest to consider more testcase in this PR or other PR:
1. use SDK to write data and read by CarbonFileInputFormat through mapreduce
2. use SDK to write data and read by CarbonFileInputFormat through spark sql
3. use SDK to write data without schema persistence, and read by CarbonFileInputFormat through spark sql
4. use SDK to write data to same folder multiple times, without schema persistence, and read by CarbonFileInputFormat through spark sql (to mimic multiple load)
5. different sorting configuration of 1-4
6. use CarbonTableOutpuFormat to write data, and read by CarbonFileInputFormat using the segment folder
7. For the partition related test, maybe need to added after #1984 is merged